packages/@react-stately/selection/src/SelectionManager.ts

```diff
@@ -275,13 +278,16 @@ export class SelectionManager implements MultipleSelectionManager {
     let addKeys = (key: Key) => {
       while (key) {
         let item = this.collection.getItem(key);
-        if (item.type === 'item') {
-          keys.push(key);
-        }
 
-        // Add child keys. If cell selection is allowed, then include item children too.
-        if (item.hasChildNodes && (this.allowsCellSelection || item.type !== 'item')) {
-          addKeys([...item.childNodes][0].key);
+        if (!this.state.disabledKeys.has(key)) {
```

Comment:
should we move this up to wrap that let item = as well?

Reply:
sounds good